### ISSUE TITLE
[Bug]: Standalone panic: runtime error: invalid memory address or nil pointer dereference when concurrent insert multi partitions

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#23539]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20230419-f14c2211
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):  pulsar  
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. Create collection -> create index -> insert 5k -> flush -> create index -> load
2. Concurrent 3 users run `concurrent_scene_insert_partition`: create a partition -> insert 2k with `ni=1000` -> drop partition
Standalone panic:

panic log:
```
[2023/04/19 10:07:10.821 +00:00] [INFO] [datacoord/server.go:865] ["flush successfully"] [segmentID=440899567108367223]
[2023/04/19 10:07:10.821 +00:00] [INFO] [datacoord/services.go:482] ["compaction triggered for segment"] [traceID=a779629e0f1ae6a71ee645df6a28a955] [nodeID=2] [collectionID=440899567108166933] [segmentID=440899567108367223]
[2023/04/19 10:07:10.821 +00:00] [INFO] [datacoord/services.go:482] ["compaction triggered for segment"] [traceID=d45bf7b3c8a287b60fe63a3c7610ff90] [nodeID=2] [collectionID=440899567108166933] [segmentID=440899567108367234]
[2023/04/19 10:07:10.821 +00:00] [INFO] [datanode/channel_meta.go:447] ["remove segments if exist"] [segmentIDs="[440899567108367223]"]
[2023/04/19 10:07:10.821 +00:00] [WARN] [datanode/channel_meta.go:770] ["cannot find segment when evictHistoryInsertBuffer"] [segmentID=440899567108367223]
[2023/04/19 10:07:10.821 +00:00] [WARN] [datanode/channel_meta.go:817] ["cannot find segment when evictHistoryDeleteBuffer"] [segmentID=440899567108367223]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x288a2f1]

goroutine 57303 [running]:
panic({0x3327e20, 0x4e9bff0})
    /usr/local/go/src/runtime/panic.go:941 +0x397 fp=0xc00407d4f8 sp=0xc00407d438 pc=0x1259157
runtime.panicmem(...)
    /usr/local/go/src/runtime/panic.go:220
runtime.sigpanic()
    /usr/local/go/src/runtime/signal_unix.go:818 +0x336 fp=0xc00407d548 sp=0xc00407d4f8 pc=0x1270176
github.com/milvus-io/milvus/internal/datanode.(*Segment).getType(...)
    /go/src/github.com/milvus-io/milvus/internal/datanode/segment.go:66
github.com/milvus-io/milvus/internal/datanode.(*ChannelMeta).new2NormalSegment(0x1?, 0xc00407d5c8?)
    /go/src/github.com/milvus-io/milvus/internal/datanode/channel_meta.go:167 +0x31 fp=0xc00407d588 sp=0xc00407d548 pc=0x288a2f1
github.com/milvus-io/milvus/internal/datanode.(*ChannelMeta).transferNewSegments(0xc00199f830, {0xc0047c3860, 0x3, 0x3599de0?})
    /go/src/github.com/milvus-io/milvus/internal/datanode/channel_meta.go:426 +0x92 fp=0xc00407d5d8 sp=0xc00407d588 pc=0x288e0d2
github.com/milvus-io/milvus/internal/datanode.flushNotifyFunc.func1.1()
    /go/src/github.com/milvus-io/milvus/internal/datanode/flush_manager.go:845 +0x169 fp=0xc00407d7a8 sp=0xc00407d5d8 pc=0x28c1a09
github.com/milvus-io/milvus/pkg/util/retry.Do({0x3b19368, 0xc000236140}, 0xc00407db10, {0xc000d603c0, 0x1, 0x1829465?})
    /go/src/github.com/milvus-io/milvus/pkg/util/retry/retry.go:40 +0x108 fp=0xc00407d968 sp=0xc00407d7a8 pc=0x20467a8
github.com/milvus-io/milvus/internal/datanode.flushNotifyFunc.func1(0xc003a7b450)
    /go/src/github.com/milvus-io/milvus/internal/datanode/flush_manager.go:818 +0xfad fp=0xc00407df78 sp=0xc00407d968 pc=0x28c132d
github.com/milvus-io/milvus/internal/datanode.(*flushTaskRunner).waitFinish(0xc0028c0780, 0xc005025860, 0xc002ecaac0)
    /go/src/github.com/milvus-io/milvus/internal/datanode/flush_task.go:205 +0xdb fp=0xc00407dfb8 sp=0xc00407df78 pc=0x28c301b
github.com/milvus-io/milvus/internal/datanode.(*flushTaskRunner).init.func1.1()
    /go/src/github.com/milvus-io/milvus/internal/datanode/flush_task.go:122 +0x2e fp=0xc00407dfe0 sp=0xc00407dfb8 pc=0x28c226e
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc00407dfe8 sp=0xc00407dfe0 pc=0x128e6a1
created by github.com/milvus-io/milvus/internal/datanode.(*flushTaskRunner).init.func1
    /go/src/github.com/milvus-io/milvus/internal/datanode/flush_task.go:122 +0xf1

goroutine 1 [chan receive, 7 minutes]:
runtime.gopark(0xc00087d728?, 0xc00087d780?, 0x13?, 0xae?, 0xc00087d780?)
    /usr/local/go/src/runtime/proc.go:361 +0xd6 fp=0xc0021c9750 sp=0xc0021c9730 pc=0x125c3b6
runtime.chanrecv(0xc000837860, 0xc00087dbe8, 0x1)
    /usr/local/go/src/runtime/chan.go:577 +0x56c fp=0xc0021c97e0 sp=0xc0021c9750 pc=0x1225aac
runtime.chanrecv1(0xc000837860?, 0xc00087ddb8?)
    /usr/local/go/src/runtime/chan.go:440 +0x18 fp=0xc0021c9808 sp=0xc0021c97e0 pc=0x12254d8
github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc002c35e58, 0x1, {0xc0007240d0?, 0xe?})
    /go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:336 +0xa6c fp=0xc0021c9e08 sp=0xc0021c9808 pc=0x2e52e4c
```

### Steps To Reproduce

```markdown
Case:

@pytest.mark.locust
    @pytest.mark.parametrize("deploy_mode", [CLUSTER, STANDALONE])
    def test_debug(self, input_params: InputParamsBase, deploy_mode):
        """
        :test steps:
            1. concurrent test and calculation of RT and QPS
        """
        data_size = "5k"

        # concurrent tasks
        concurrent_tasks = [
            ConcurrentParams.params_scene_insert_partition(weight=10, data_size='2k', ni=1000, with_flush=False,
                                                           timeout=600),
            ConcurrentParams.params_release(weight=2, timeout=600)
        ]

        # concurrent params
        default_case_params = ConcurrentParams().params_scene_concurrent(
            concurrent_tasks, concurrent_number=[3], during_time="2m", interval=10, dim=128, dataset_size=data_size,
            ni_per=5000)

        self.concurrency_template(input_params=input_params, cpu=dp.min_cpu, mem=8, deploy_mode=deploy_mode,
                                  old_version_format=False,
                                  case_callable_obj=ConcurrentClientBase().scene_concurrent_locust,
                                  default_case_params=default_case_params)
```
```


### Milvus Log

devops cluster, chaos-testing ns pods:
```
daoooo-ohaia-etcd-0                                               1/1     Running            0                 57m
daoooo-ohaia-milvus-standalone-687c4fcd7b-98gf8                   1/1     Running            6 (7m2s ago)      57m
daoooo-ohaia-minio-9f9b9747d-696mz                                1/1     Running            0                 57m
```
standalone_pre.log:
[s_pre.log](https://github.com/milvus-io/milvus/files/11272101/s_pre.log)


### Anything else?

_No response_

**Comment by congqixia** [comment]
@ThreadDao Patch pr merged. Could you please verify?
/unassign
/assign @ThreadDao

**Comment by ThreadDao** [comment]
fixed `master-20230421-ccba1334`
